Test Failure: TestController_GetIstioServiceAccounts

https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17142/racetest_istio/458

```
--- FAIL: TestController_GetIstioServiceAccounts (0.23s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x20da310]
 goroutine 129 [running]:
testing.tRunner.func1(0xc000688000)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x22a3be0, 0x365fa80)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
istio.io/istio/pilot/pkg/serviceregistry/kube/controller.(*Controller).GetIstioServiceAccounts(0xc000688100, 0x0, 0xc000151c58, 0x0, 0x0, 0x0, 0x18, 0x3)
	/home/prow/go/src/istio.io/istio/pilot/pkg/serviceregistry/kube/controller/controller.go:803 +0x4f0
istio.io/istio/pilot/pkg/serviceregistry/kube/controller.TestController_GetIstioServiceAccounts(0xc000688000)
	/home/prow/go/src/istio.io/istio/pilot/pkg/serviceregistry/kube/controller/controller_test.go:677 +0x1b1c
testing.tRunner(0xc000688000, 0x258eb60)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652
FAIL	istio.io/istio/pilot/pkg/serviceregistry/kube/controller	1.453s
=== RUN   TestMemoryServices
--- PASS: TestMemoryServices (0.02s)
PASS
ok  	istio.io/istio/pilot/pkg/serviceregistry/memory	1.076s
```

--- a/pilot/pkg/serviceregistry/kube/controller/controller_test.go
+++ b/pilot/pkg/serviceregistry/kube/controller/controller_test.go
@@ -639,6 +639,7 @@ func TestController_GetIstioServiceAccounts(t *testing.T) {
 			annotation.AlphaKubernetesServiceAccounts.Name: k8sSaOnVM,
 			annotation.AlphaCanonicalServiceAccounts.Name:  canonicalSaOnVM},
 		[]int32{8080}, map[string]string{"app": "prod-app"}, t)
+	<-fx.Events
 	createService(controller, "svc2", "nsA", nil, []int32{8080}, map[string]string{"app": "staging-app"}, t)
 	<-fx.Events
 

